code/game/turfs/simulated/floor_icon.dm

```diff
@@ -84,6 +84,11 @@ var/list/flooring_cache = list()
 				continue
 			F.update_icon()
 
+	// Show 'ceilingless' overlay.
+	var/turf/above = GetAbove(src)
+	if(above && isopenspace(above) && !istype(src, /turf/simulated/floor/outdoors)) // This won't apply to outdoor turfs since its assumed they don't have a ceiling anyways.
+		overlays |= image(icon = 'icons/turf/open_space.dmi', icon_state = "no_ceiling", layer = TURF_LAYER + 0.02)
```

Comment:
I'd suggest caching this image so you don't need to make a new one for literally every turf. Just have it stored in a global somewhere. You could also set a `plane` on this image to make it so you can toggle the visibility with a verb or just wearing mesons or whatever.

In the case of planes, I can help you set that up.

Reply:
Will look into seeing if I can put it on a plane for meson vision or something with Aronai's help.